docs/source/gateway.mdx

```diff
@@ -57,41 +57,137 @@ To learn how to compose your supergraph schema with the Rover CLI, see the [Fede
 
 On startup, the gateway processes your `supergraphSdl`, which includes routing information for your subgraphs. It then begins accepting incoming requests and creates query plans for them that execute across one or more subgraphs.
 
-### Composing with `serviceList`
+### Updating the supergraph schema
+
+In the above example, we provide a _static_ supergraph schema to the gateway. This approach requires the gateway to be restarted in order to update the supergraph schema. This is less than ideal for many applications, so we also provide the ability to update the supergraph schema dynamically.
+
+```js:title=index.js
+const { ApolloServer } = require('apollo-server');
+const { ApolloGateway } = require('@apollo/gateway');
+const { readFileSync } = require('fs');
+
+const supergraphSdl = readFileSync('./supergraph.graphql').toString();
+
+let supergraphUpdate;
+const gateway = new ApolloGateway({
+  async supergraphSdl({ update }) {
+    // `update` is a function which we'll save for later use
+    supergraphUpdate = update;
+    return {
+      supergraphSdl,
+    }
+  },
+});
+
+// Pass the ApolloGateway to the ApolloServer constructor
+const server = new ApolloServer({
+  gateway,
+});
+
+server.listen().then(({ url }) => {
+  console.log(`🚀 Server ready at ${url}`);
+});
+```
+
+There are a few things happening here, let's take a look at each of them individually.
+
+Note that `supergraphSdl` is now an `async` function. This function is only called once when `ApolloServer` initializes the gateway. This function has two primary responsibilities:
+1. Provide us access to the `update` function, which we'll use to update the supergraph schema.
```

Comment:
The text between the two bullets isn't exactly parallel. #2 is what the function does (it returns a value).  #1 feels more like what ApolloGateway does — it provides us access to `update`.  For better parallelism shouldn't this line be more like "receive the `update` function"?

Reply:
e9e191ef